pom.xml

```diff
@@ -119,7 +119,7 @@
     <mesos.classifier>shaded-protobuf</mesos.classifier>
     <slf4j.version>1.7.5</slf4j.version>
     <log4j.version>1.2.17</log4j.version>
-    <hadoop.version>1.0.4</hadoop.version>
+    <hadoop.version>2.4.0</hadoop.version>
```

Comment:
filed this just before I finished composing [SPARK-5115](https://issues.apache.org/jira/browse/SPARK-5115) because I wanted to a PR ID to reference there :)

tl;dr is that IntelliJ doesn't resolve YARN imports because the spark's `yarn`-module POM inherits `hadoop.version` from the parent POM, and YARN didn't exist at `1.0.4`